controllers/storagecluster/reconcile.go

```diff
@@ -950,7 +950,7 @@ func getUnsupportedClientsCount(r *StorageClusterReconciler, namespace string) (
 		if scList.Items[idx].Status.Client != nil {
 			clientVersion, err := semver.Make(scList.Items[idx].Status.Client.OperatorVersion)
 			if err == nil {
-				if providerVersion.Major != clientVersion.Major || providerVersion.Minor <= clientVersion.Minor {
+				if !(providerVersion.Major == clientVersion.Major || providerVersion.Minor <= clientVersion.Minor) {
```

Comment:
There is a bug here, If you are to move the not (!) outside the || need to become a &&
But why not just change <= to >? It will be logically equivalent and a smaller change as
!(A && B) = !A || !B